src/components/MemberArea/MemberArea.js

```diff
@@ -42,11 +44,41 @@ function MemberArea({ onOpenModal }) {
     });
   }, [currentUser, openProfile]);
 
+  useEffect(() => {
+    if(currentUser){
+      const isAvailable = isMentorAvailable();
+      setIsAvailable(isAvailable);
+    }
+  }, [currentUser]);
+
+  useEffect(() => {
+    if(currentUser){
+      const cachedAvailabilityState = isMentorAvailable();
+        if(cachedAvailabilityState !== isAvailable) {
+            updateMentorAvailability(isAvailable).then((isSuccessful)=>{
+              if(!isSuccessful){
+                setIsAvailable((isAvailable) => !isAvailable);
```

Comment:
(Like I mention above) You can use [`updateUser`](https://github.com/Coding-Coach/find-a-mentor/blob/c030fe16bcae966c3bea547af59bc9283985300e/src/context/userContext/UserContext.js#L7)